hack/gha-reversemap.sh

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Purpose: Harmonise '$GITHUB_WORKFLOWS_PATH' and the reversemap '$REVERSEMAP_FILE'
+
+# Usage: see help function
+# Working directory must be the project root
+
+# Define the required version of yq
+GITHUB_TOKEN=
+GITHUB_WORKFLOWS_PATH="./.github/workflows"
+REVERSEMAP_FILE=".gha-reversemap.yml"
+YQ_REQUIRED_VERSION="v4.45.1"
+GIT_COMMITSHA_LENGTH=40
+
+ERR_YQ_DOWNLOAD_FAILED=50
+ERR_YQ_NOT_INSTALLED=60
+ERR_GITHUB_TOKEN_INVALID=70
+ERR_ARCH_UNSUPPORTED=80
+
+help() {
+    cat <<EOF
+Harmonise '$GITHUB_WORKFLOWS_PATH' and the reversemap '$REVERSEMAP_FILE'
+
+Usage:
+    $0  <operation> [ARGUMENTS]
+
+Example:
+    # Update actions/checkout to its latest release
+    $0  update actions/checkout
+
+Operations:
+    apply-reversemap        [...WORKFLOW_FILE]          -   Apply the commit sha of every actions listed in the reversemap 
+                                                            as the version to use in all github workflows
+
+    update-action-version   ACTION_REF [...ACTION_REF]  -   Update the version of the given action reference within the reversemap
```

Comment:
I think that `update-action-version ACTION_REF...` would get the message across.